src/toeplitz.jl

```diff
@@ -0,0 +1,143 @@
+# General Toeplitz matrix
+"""
+    Toeplitz
+
+A Toeplitz matrix.
+"""
+struct Toeplitz{T<:Number} <: AbstractToeplitz{T}
+    vc::AbstractVector{T}
```

Comment:
These need to be templated out. We can also drop the condition that `T<:Number`. I.e. we want
```julia
struct Toeplitz{T, VC<:AbstractVector{T}, VR<:AbstractVector{T}} <: AbstractToeplitz[T}
    vc::VC
    vr::VR
    ...
end
```

Reply:
Got it.

Reply:
What does "template out" mean?